backend/src/packages/users/user.entity.ts

```diff
@@ -0,0 +1,88 @@
+import { type Entity } from '#libs/interfaces/interfaces.js';
+
+class UserEntity implements Entity {
+  private 'id': number | null;
+
+  private 'email': string;
+
+  private 'passwordHash': string;
+
+  private 'passwordSalt': string;
+
+  private constructor({
+    id,
+    email,
+    passwordHash,
+    passwordSalt,
+  }: {
+    id: number | null;
+    email: string;
+    passwordHash: string;
+    passwordSalt: string;
+  }) {
+    this.id = id;
+    this.email = email;
+    this.passwordHash = passwordHash;
+    this.passwordSalt = passwordSalt;
+  }
+
+  public static initialize({
+    id,
+    email,
+    passwordHash,
+    passwordSalt,
+  }: {
+    id: number;
+    email: string;
+    passwordHash: string;
+    passwordSalt: string;
+  }): UserEntity {
```

Comment:
I think it is not needed here, we will only use them once and we will need to define a lot of such types, at least 5 (but can be much more). So it will be hard to navigate this file